src/lib/string.ts

```diff
@@ -64,3 +64,39 @@ export function truncate(value: string | undefined, maxLength: number, suffix =
 
   return `${value.slice(0, maxLength)}${suffix}`;
 }
+
+/**
+ * Validation of social insurance number with checking the checksum
+ * Validation according to https://www.sozialversicherungsnummer.ch/aufbau-neu.htm
+ * @param socialInsuranceNumber The social insurance number to check
+ * Must be in one of the following formats:
+ * - "756.XXXX.XXXX.XX" with dots as seperators
+ * - "756XXXXXXXXXX" with digits only
+ * @returns The result if the social insurance number is valid or not
+ */
+export function isValidSwissSocialSecurityNumber(socialInsuranceNumber: string): boolean {
+  if (isNullOrWhitespace(socialInsuranceNumber)) {
+    return false;
+  }
+
+  const socialInsuranceNumberWithDots = new RegExp(/^756.\d{4}.\d{4}.\d{2}$/);
+  const socialInsuranceNumberWithoutDots = new RegExp(/^756\d{10}$/);
+
+  if (!socialInsuranceNumberWithDots.test(socialInsuranceNumber) && !socialInsuranceNumberWithoutDots.test(socialInsuranceNumber)) {
+    return false;
+  }
+  const compactNumber = socialInsuranceNumber.replaceAll(".", "");

```

Comment:
also here would be nice to have a block comment to explain in few words (or bullet point list) the validation logic